### PR TITLE
refactor(dom): Add util for finding slotted elements

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #784 

## Summary

After writing the tests, it became evident that the API could be simpler (i.e., `all` flag). Would it be preferable to have 2 utils `getSlotted` & `getSlottedAll` or just accept a 3rd boolean (instead of an object) for `all`?

### CODE REVIEW

**Comment by driskull** [comment]
I like the 3rd boolean option rather than two methods. Or we just always do all and they can test for an empty array?

**Comment by jcfranco** [comment]
> Or we just always do all and they can test for an empty array?

I kind of like this better. I'll make some changes.

**Comment by driskull** [comment]
> I kind of like this better. I'll make some changes.

Sounds good!

File: src/components/utils/dom.spec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        document.body.firstChild.remove();
+      }
+    });
+
+    it("returns elements with matching slot name", () => {
+      expect(getSlotted(document.body, slotNameMultiple)).toHaveLength(2);
+      expect(getSlotted(document.body, slotNameSingle)).toHaveLength(1);
+    });
+
+    it("returns null when no results", () => expect(getSlotted(document.body, "non-existent-slot")).toHaveLength(0));
```
Comment (driskull) [inline src/components/utils/dom.spec.ts]: nitpick: rename to "returns empty array when no results"?

File: src/components/utils/dom.spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+      }
+
+      customElements.define("slot-test", SlotTest);
+    });
+
+    beforeEach(() => {
+      document.body.innerHTML = `
+      <slot-test>
+        <h2 slot=${slotNameMultiple}>😃</h2>
```
Comment (driskull) [inline src/components/utils/dom.spec.ts]: :)

File: src/components/utils/dom.spec.ts (hunk 1 above)
Comment (jcfranco) [reply src/components/utils/dom.spec.ts]: D-oh. Good catch.

**Pull request merged**